app/components/chart_component.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ChartComponent < ViewComponent::Base
+  renders_one :title
+  renders_one :subtitle
+  renders_one :header
+  renders_one :footer
+
+  attr_reader :school, :chart_type, :chart_config, :analysis_controls, :no_zoom, :html_class
+
+  include ChartHelper
+
+  def initialize(chart_type: '', school: nil, chart_config: {}, analysis_controls: false, no_zoom: true, html_class: 'analysis-chart')
```

Comment:
It looks like we'll end up with lots of repeated calls to `create_chart_config(@school, :baseload_lastyear)` across charts. And both of the school and the chart type are already passed as parameters. Could we just let the component call `create_chart_config` to avoid repeated code. As I think in most cases that's all we'll want.

We could allow `chart_config` to be provided as an optional parameter, if we needed something particularly custom?

Reply:
Can you also check the defaults here, as I think in most cases we have both analysis controls and allow zoom? Can probably check across what we're doing in a few of the partials across the app currently?